src/utils/handleOptions.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Evaluates the value of a moldObject property.
+ * If the value is a function, it is called with the filtered options object as the its parameter.
+ * Otherwise, the value is returned as is.
+ * @param {object} filteredObject - the current filtered options object
+ * @param {any} value - the value to evaluate
+ * @returns {any} the evaluated value
```

Comment:
Just for reference.  Not suggesting we do this.  It's probably overkill but I did write an accurate type.  

I tried writing the types using JSDoc but I couldn't get the `@template T` working properly.  So I wrote it in TypeScript.

```
function handleOptions<T extends Record<string, any>>(userObject: T, moldObject: MoldForOptions<T>, modelName: string) {
```

```
interface MoldConfig<D> {
    /**
     * A string defining the correct type.
     */
    type: "number" | "enum" | "boolean" | "string" | "object" | "undefined";
    /**
     * The default value in case the user does not provide a value or provides an erroneous value for the option.
     */
    default: D;
    /**
     * A boolean defining whether the key should be ignored. Defaults to false. Useful when set to a dynamically evaluated value.
     */
    ignore?: boolean;
    /**
     * A number defining the minimum value.
     */
    min?: number;
    /**
     * A number defining the maximum value.
     */
    max?: number;
    /**
     * A boolean defining whether the value should be an integer. Defaults to `false`.
     */
    integer?: boolean;
    /**
     * An array defining a list of valid values.
     */
    enums?: Array<D>;
    /**
     * A boolean defining whether to checks the user value against the enum list case-insensitively. Defaults to `true`.
     */
    caseInsensitive?: boolean;
}

type MoldProperty<PropertyType, OptionsType> = {
    [K in keyof MoldConfig<PropertyType>]?: MoldConfig<PropertyType>[K] | ((options: OptionsType) => MoldConfig<PropertyType>[K])
}

type MoldForOptions<OptionsType> = {
    [K in keyof OptionsType]: MoldProperty<OptionsType[K], OptionsType>;
}
```
[The rest of the types](https://tsplay.dev/w2g49N)

It's a complicated mess 👎 but you do get good Intellisense 👍.

Hovering over a property of the options shows the description of that option, based on what I defined in the `HandposeOptions` interface.
![image](https://github.com/ml5js/ml5-next-gen/assets/28965286/e85e8992-e779-43aa-a4ac-1fa8a3409b51)

Hovering over a field like `default` shows the description of the field and possible values (again, based on `HandposeOptions`).
![image](https://github.com/ml5js/ml5-next-gen/assets/28965286/04822b15-785e-4116-bb7a-6c98a2048d46)

Incorrect configurations give an error.
![image](https://github.com/ml5js/ml5-next-gen/assets/28965286/91c84552-d1e1-4fbc-b484-7b78de2ed21d)

Incorrect options give an error.
![image](https://github.com/ml5js/ml5-next-gen/assets/28965286/6e87a117-9db7-40f5-b7e9-dfc40b452506)

Reply:
> I tried writing the types using JSDoc but I couldn't get the `@template T` working properly. So I wrote it in TypeScript.

Turns out it was really simple.  I was trying to put a comment on the same line as the `T` but that was breaking it.

Something like this is more accurate, but also more confusing to anyone who isn't me so idk.
```
/**
 * Evaluates the value of a moldObject property.
 * If the value is a function, it is called with the filtered options object as its parameter.
 * Otherwise, the value is returned as is.
 * @template T
 * @param {object} filteredObject - the current filtered options object
 * @param {T | ((options: object) => T)} value - the value to evaluate
 * @returns {T} the evaluated value
 */
function evaluate(filteredObject, value) {
```